src/libraries/shims/ApiCompat.proj

```diff
@@ -7,7 +7,7 @@
 
   <PropertyGroup>
     <ApiCompatTarget>$(TargetFramework)</ApiCompatTarget>
-    <ApiCompatTarget Condition="$(BuildTargetFramework.Contains('netcoreapp'))">netcoreapp</ApiCompatTarget>
+    <ApiCompatTarget Condition="'$(TargetsNetCoreApp)' == 'true'">netcoreapp</ApiCompatTarget>
```

Comment:
TargetsNetCoreApp wont be set here as we are just setting the TargetFramework on line5.